### PR TITLE
Different access required for evaluating new alerts vs existing alerts.

### PR DESCRIPTION
**Pull request by jacksonargo**
New alerts require write permission to evaluate.
Existing alerts require read permission to evaluate.

https://cortex-data.slack.com/archives/C031NQQNDPX/p1676496715238079
https://cortexdata.atlassian.net/browse/TE-1336

### CODE REVIEW

**Comment by jacksonargo** [comment]
Waiting until https://github.com/startreedata/thirdeye/pull/961 is merged, since it will create merge conflicts w/in /evaluate handler.

File: thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java [hunk 1]
```diff
@@ -236,15 +236,19 @@ public Response evaluate(
...
-    // see TE-1172 - ensure the dto passed to the authManager is complete, not only a reference {id:1234}
-    final AlertDTO alertDTO = optional(alert.getId()).map(this::get).orElseGet(() -> toDto(alert));
-    authorizationManager.ensureCanEvaluate(principal, alertDTO);
-
-    final AlertEvaluationApi results = alertEvaluator.evaluate(request);
+    final AlertApi alertApi = ensureExists(request.getAlert())
+        .setOwner(new UserApi().setPrincipal(principal.getName()));
+
+    AlertEvaluationApi results;
+    if (alertApi.getId() != null) {
```
Comment (suvodeep-pyne) [inline thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java:243]: This introduces execution complexity. I'd be interested to know what is the longer term motivation for this change. 

My understanding was that we just need to decide on how the auth check is being performed.

Comment (jacksonargo) [reply thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java:243]: I was trying to be more explicit about what is considered _new alert eval_ vs _existing alert eval_.
The code that checks for the alert id and loads it from the db is fairly subtle.

Comment (jacksonargo) [reply thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java:243]: I don't feel too strongly about this refactor, tho. I'm fine with only checking for the alert id and no other changes.

**Review by suvodeep-pyne — APPROVED** [review]
LGTM

**Pull request merged**